### ISSUE TITLE
EntityTypeBuilder - the order of calling HasKey and HasOne methods matters

### ISSUE THREAD

**Issue by jirikanda** [dotnet/efcore#14073]
Starting from version 2.2.0 when HasKey is called before HasOne, the later call of HasOne method defines a new shadow property. In 2.1.4 version this does not happen.

### Steps to reproduce
```c#
	public class Member
	{
		public int Id { get; set; }
		public List<Membership> LeftMemberships { get; set; }
		public List<Membership> RightMemberships { get; set; }
	}

	public class Membership
	{
		public int LeftMemberId { get; set; }
		public Member LeftMember { get; set; }

		public int RightMemberId { get; set; }
		public Member RightMember { get; set; }
	}

	public class ApplicationDbContext : DbContext
	{
		protected override void OnModelCreating(ModelBuilder modelBuilder)
		{
			base.OnModelCreating(modelBuilder);

			// in version 2.1.4 this line can be here
			// since version 2.2.0 if this method is here, Membership has declared 3 properties (one shadow)
			// Workaround to 2.2.0: Move HasKey method call below and remove this call
			modelBuilder.Entity<Membership>().HasKey(membership => new { membership.LeftMemberId, membership.RightMemberId });

			modelBuilder.Entity<Membership>().HasOne(membership => membership.LeftMember).WithMany(member => member.RightMemberships); // members point to the right, memberships point to the left
			modelBuilder.Entity<Membership>().HasOne(membership => membership.RightMember).WithMany(member => member.LeftMemberships); // etc.

			// Workaround to 2.2.0: Move HasKey method call here and remove previous call
			//modelBuilder.Entity<Membership>().HasKey(membership => new { membership.LeftMemberId, membership.RightMemberId });
		}

		protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
		{
			base.OnConfiguring(optionsBuilder);

			optionsBuilder.UseInMemoryDatabase(nameof(ApplicationDbContext));
		}
	}

	[TestClass]
	public class ModelTest
	{
		[TestMethod]
		public void Model_Membership_HasExactlyTwoProperties()
		{
			ApplicationDbContext dbContext = new ApplicationDbContext();
			Assert.AreEqual(2, dbContext.Model.FindEntityType(typeof(Membership)).GetProperties().Count());
		}
	}
```

### Further technical details
EF Core version: 2.2.0
Database Provider: Microsoft.EntityFrameworkCore.InMemory

**Comment by AndriySvyryd** [comment]
Confirmed regression. But since there's a workaround we will not fix this in a patch unless it affecs a large portion of users.

**Comment by AndriySvyryd** [comment]
Doesn't repro on 3.0. Probably fixed by 3c60068fe1dedd36cf9e43e5398653601ad17748
